connector/avro/src/test/scala/org/apache/spark/sql/avro/AvroSuite.scala

```diff
@@ -921,6 +921,39 @@ abstract class AvroSuite
     }
   }
 
+  test("SPARK-49082: Widening type promotions in AvroDeserializer") {
```

Comment:
Thanks for implementing this (I posted the original ticket :D)!

Is there a plan to convert Date -> TimestampNTZ (Parquet [PR](https://github.com/apache/spark/pull/44368/files#diff-e067188f85ea958fe1fdfc6f83466def9ca962ebd39ca9a86aaee1ed3110c045R102)) and Int -> Decimal (Parquet [PR](https://github.com/apache/spark/pull/44803))?

Reply:
Yes, I will seperate another two PR.